caffe2/core/export_c10_op_to_caffe2.h

```diff
@@ -1,13 +1,16 @@
 #pragma once
 
+#include <c10/macros/Macros.h>
+#include <c10/util/Registry.h>
+#include "caffe2/core/operator.h"
+
 // TODO Also register c10 operators on mobile
-#if !defined(CAFFE2_IS_XPLAT_BUILD)
+#if !defined(CAFFE2_IS_XPLAT_BUILD) && !defined(C10_MOBILE)
```

Comment:
Are there instances where we only consider CAFFE2_IS_XPLAT_BUIILD or C10_MOBILE? Could we delete one of these two in the future?

I remember @ljk53 was talking about consolidating the mobile build flags.

Reply:
It looks like the answer is that, most of the time, we consider both as a pair. The only place this is not true is:

```
aten/src/ATen/core/op_registration/op_registration.cpp
2:#if !defined(CAFFE2_IS_XPLAT_BUILD)
                                                              
aten/src/ATen/core/op_registration/op_registration.h          
10:#if !defined(CAFFE2_IS_XPLAT_BUILD)                                      
99:        #if defined(CAFFE2_IS_XPLAT_BUILD)

aten/src/ATen/core/EnableNamedTensor.h
9:#if !defined(CAFFE2_IS_XPLAT_BUILD) && (!defined(C10_MOBILE) || defined(FEATURE_TORCH_MOBILE))
```

So agreed here.

Reply:
I think semantically CAFFE2_IS_XPLAT_BUILD should be equivalent to "C10_MOBILE && ! FEATURE_TORCH_MOBILE" - it's xplat mobile build (thus C10_MOBILE) and caffe2 only as in current production (thus !FEATURE_TORCH_MOBILE). 

One argument @smessmer had to keep CAFFE2_IS_XPLAT_BUILD was that xplat has broader meaning than mobile, but I think the substantial difference is whether to build full torch (FEATURE_TORCH_MOBILE).

Reply:
I don't really have a stone in the game, except that I don't want to make too many more changes to this diff as it stands right now :)